### PR TITLE
Move config generator code to tools

### PR DESCRIPTION
**Pull request by chizhg**
**What this PR does, why we need it**:
Move config generator code to tools/config-generator

### CODE REVIEW

File: tools/config-generator/README.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# config-generator
+
+config-generator is a tool that takes a meta config file (e.g. ../../config/prow/config_knative.yaml)
```
Comment (chaodaiG) [inline tools/config-generator/README.md]: Thank you for this README, I think we should also mention where templates are. Related, should we move templates here?

Comment (chizhg) [reply tools/config-generator/README.md]: Done.

**Pull request merged**